x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/output/AutodetectResultProcessor.java

```diff
@@ -310,6 +325,21 @@ void processResult(AutodetectResult result) {
         }
     }
 
+    private Annotation createModelSnapshotAnnotation(ModelSnapshot modelSnapshot) {
+        assert modelSnapshot != null;
+        Date currentTime = new Date(clock.millis());
+        return new Annotation(
+            Messages.JOB_AUDIT_SNAPSHOT_STORED,
+            currentTime,
+            XPackUser.NAME,
+            modelSnapshot.getTimestamp(),
+            modelSnapshot.getLatestRecordTimeStamp(),
```

Comment:
I think it could be confusing if the annotation spans a time range.  Is it possible to set both the start and end time to be the model snapshot's timestamp?  Or does it break the UI if the start and end timestamps are identical?